File: backend/java/localpos-user-service/src/main/java/com/pds/localpos/userservice/controller/AuthController.java
package com.pds.localpos.userservice.controller;

import com.pds.localpos.userservice.dto.request.LoginRequest;
import com.pds.localpos.userservice.dto.response.UserResponse;
import com.pds.localpos.userservice.service.AuthService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/v1/auth")
@RequiredArgsConstructor
@Slf4j
public class AuthController {

    private final AuthService authService;

    @PostMapping("/validate")
    public ResponseEntity<UserResponse> validateUser(@Valid @RequestBody LoginRequest request) {
        log.info("POST /auth/validate - Authenticating user: {}", request.username());
        UserResponse user = authService.validateCredentials(request);
        log.info("Authentication successful for user: {}", request.username());
        return ResponseEntity.ok(user);
    }
}

File: backend/kotlin/localpos-user-service/src/main/kotlin/com/pds/localpos/userservice/controller/AuthController.kt
package com.pds.localpos.userservice.controller

import com.pds.localpos.userservice.dto.AuthRequestDTO
import com.pds.localpos.userservice.dto.AuthResponseDTO
import com.pds.localpos.userservice.service.AuthenticationService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/auth")
class AuthController(
    private val authenticationService: AuthenticationService
) {

    @PostMapping("/login")
    fun login(@RequestBody request: AuthRequestDTO): ResponseEntity<AuthResponseDTO> {
        val token = authenticationService.authenticate(request.username, request.password)
        return ResponseEntity.ok(AuthResponseDTO(token))
    }
}
